### DISCUSSION TITLE
Scrollable Sidebar for the Client

### DISCUSSION THREAD

**Discussion by colossal-wish** [runelite/runelite#14953] [Ideas]
I'd like to request a scrollable sidebar for the client

I have a lot of plugins and I feel runelite is at its best when they're all enabled, and I enjoy the convenience of being able to just click over when I need one of those plugins. The problem occurs when I began too many of them, when you have too many, the sidebar ends up extending beyond the limits of the client, and this is all in fullscreen. If the client is at the default, smallest size, it cuts 75 percent of the sidebar plugins off, making them completely inaccessible.

I propose a scrollable sidebar to appear when you have plugins that go out of the client, alternatively adding new columns would probably work as well, though I think just a scrollbar would look the best.

![image](https://user-images.githubusercontent.com/45466145/145611972-cd49fd6a-1387-4fb2-b34a-cbf14dcd7f15.png)
this showcases some of the sidebar plugins, the screenshot sidebar icon is completely cut off.

![image](https://user-images.githubusercontent.com/45466145/145611738-6738b67a-3673-4524-9013-bea4820ed957.png)
this is an example of the majority of the plugins being cut off at the default window size.

**Comment by rbbi** [comment]
Hi @cmhcreater 

Something like this?

https://user-images.githubusercontent.com/31489752/148472618-11cd69b2-843e-4dd2-92cb-37d1053c293e.mp4

**Comment by colossal-wish** [comment]
> Hi @cmhcreater
> 
> Something like this?
> 
>  v1.mp4

My god thats perfect

**Comment by colossal-wish** [comment]
@rbbi any updates on that!

Like if it'll be a plugin or anything?

**Comment by rmcleo01** [comment]
Hi @cmhcreater , addition of a scrollable sidebar was discouraged so I abandoned the idea. Check discord for more info.

**Comment by Nightfirecat** [comment]
While the issue of running out of sidebar space is something we understand, a scrolling sidebar is not something we will implement. After discussion on Discord, we agreed that we may be ok with wrapping icons to a second (and third, etc.) column to allow space for them.

With that said we still do not have a finalized design, keeping in mind that the icons drawn in the titlebar with custom chrome must also be handled when custom chrome is disabled, so we are not looking to implement this until said design has been thought through more completely.

**Reply by colossal-wish** [reply]
Sounds good anything works, I just wanna use all of my icons, and im not exactly sure if you're referring to the custom bordering for runelite vs the default windows one, but if I understand correctly, my issue still occurred even with the runelite bordering

**Reply by hawkins** [reply]
Would love to see some way to have more icons without losing access to them. I often play with windows made as small as possible to be able to watch a video behind it or such, but I then lose the ability to click on sidebar buttons like Notes that fall pretty far to the end of the list, even though I frequently disable plugins I don't need that day. I also have to go fullscreen to see the screenshot button because it falls victim to the same issue, even though it's fixed to the bottom of the panel.

**Reply by EpikPwner** [reply]
I also play with a small window and having to resize the window just to select the quest helper and then resize it again is really annoying. Would love to finally have a solution to this issue. Was writing up a new post with this suggestion and sad to see it had already been posted several years ago and there's still no resolution.

**Reply by NightmareSHW** [reply]
What about a menu for re-ordering/organizing the sidebar? It could be something similar to how you can drag/remove skills in the XP Tracker plugin.

**Reply by rbbi** [reply]
It looks like this wrapping behaviour has now been implemented:

![image](https://github.com/runelite/runelite/assets/31489752/f67ce39c-435e-42c3-a345-3fa83d0dd183)

**Comment by 369osrs** [comment]
Please make the scroll bar, 2 columns does not sit right
